src/extension.ts

```diff
@@ -221,4 +221,130 @@ export function activate(
             )
         }
     }
+
+    // Experimental: file decorations
+    if ((sourcegraph as any)?.app?.registerFileDecorationProvider) {
+        interface FileDecoration {
+            uri: string
+            after?: {
+                contentText: string
+                color: string
+                hoverMessage?: string
+            }
+            meter?: {
+                value: number
+                min?: number
+                max?: number
+                low?: number
+                high?: number
+                optimum?: number
+                hoverMessage?: string
+            }
+        }
+
+        context.subscriptions.add(
+            (sourcegraph as any).app.registerFileDecorationProvider({
+                provideFileDecorations: async ({
+                    files,
+                    uri,
+                }: {
+                    uri: string
+                    files: {
+                        uri: string
+                        isDirectory: boolean
+                        name: string
+                        path: string
+                    }[]
+                }) => {
+                    const { repo, rev } = resolveDocumentURI(uri)
+                    const apiParams = codecovParamsForRepositoryCommit({ repo, rev }, sourcegraph)
+
+                    // Fetch commit coverage to get ratio for files, tree coverage to get ratio for directories
+                    const nonDirectoryFiles = files.filter(file => !file.isDirectory)
+                    const directories = files.filter(file => file.isDirectory)
+                    const [commitCoverage, ...treeCoverages] = await Promise.allSettled([
+                        getCommitCoverage(apiParams),
+                        ...directories.map(async (dir, i) => {
+                            const treeCoverage = await getTreeCoverage({ ...apiParams, path: dir.path })
+                            return { treeCoverage, directory: directories[i] }
+                        }),
+                    ])
+
+                    // Iterate over files and get value from commit coverage to construct file decorations
+                    const fileDecorations: FileDecoration[] = []
+
+                    if (commitCoverage.status === 'fulfilled' && commitCoverage.value) {
+                        const { files: reportFiles } = commitCoverage.value.commit.report
+                        for (const file of nonDirectoryFiles) {
+                            const report = reportFiles[file.path]
+                            if (!report) {
+                                continue
+                            }
+
+                            const ratio = parseInt(report.t.c, 10)
+
+                            fileDecorations.push({
+                                uri: file.uri,
+                                after: {
+                                    contentText: `${ratio}%`,
```

Comment:
Maybe use `toFixed()` to make sure there are not super long fractions

Reply:
I'm already using `parseInt` above. Should we show a couple digits after the decimal point? I opted to show integers for readability, but maybe users want more precision

Reply:
Oh you're right, my bad.